ci-operator/step-registry/baremetalds/assisted/setup/baremetalds-assisted-setup-commands.sh

```diff
@@ -101,6 +101,13 @@ echo "export SERVICE=${ASSISTED_SERVICE_IMAGE}" >> /root/config
 echo "export AGENT_DOCKER_IMAGE=${ASSISTED_AGENT_IMAGE}" >> /root/config
 echo "export CONTROLLER_IMAGE=${ASSISTED_CONTROLLER_IMAGE}" >> /root/config
 echo "export INSTALLER_IMAGE=${ASSISTED_INSTALLER_IMAGE}" >> /root/config
+# Most jobs and tests don't require this image, so we allow as optional
+# This is required due to the nounset flag
+PROVIDER_IMAGE=${PROVIDER_IMAGE:-}
+if [ ! -z ${PROVIDER_IMAGE} ];
+then
+  echo "export PROVIDER_IMAGE=${PROVIDER_IMAGE}" >> /root/config
```

Comment:
assuming we sometimes put here the wrong image,
can we check it before? something like:
```
image_name=$(podman image inspect ${PROVIDER_IMAGE} -f "{{.RepoTags}}")
if [[ ${image_name} ~= "cluster-api-provider-agent" ...
```

Reply:
I'll update the if to check if statement too check:
if [ "${PROVIDER_IMAGE}" != "${CONTROLLER_IMAGE}" ]